src/analysis/timeline/timelineCredView.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+
+import {sum} from "d3-array";
+import sortBy from "lodash.sortby";
+import stringify from "json-stable-stringify";
+import * as NullUtil from "../../util/null";
+import {TimelineCred} from "./timelineCred";
+import {type Interval, weekIntervals} from "./interval";
+import {type Node, type NodeAddressT, NodeAddress} from "../../core/graph";
+
+/**
+ * A Graph Node wrapped with cred information.
+ */
+export type CredNode = {|
+  // The Graph Node in question
+  +node: Node,
+  // The total aggregated cred. (Summed across every interval).
+  +total: number,
+  // The timeline sequence of cred (one score per interval).
+  +cred: $ReadOnlyArray<number>,
+|};
+
+/**
+ * TimelineCredView is a subset view of a TimelineCred for a user
+ * specified time interval. TimelineCredView will extend this interval
+ * to be contained in the base intervals in TimelineCred
+ */
+export class TimelineCredView {
+  _inputInterval: Interval;
+  _viewIntervals: Interval[];
+  _timelineCred: TimelineCred;
+
+  // We are displaying a range of the intervals in the base TimelineCred
+  // We track the start index and end index of the range, so we can efficiently
+  // slice the cred
+  _startIntervalIndex: number;
+  _endIntervalIndex: number;
+
+  constructor(timelineCred: TimelineCred, inputInterval: ?Interval) {
+    this._timelineCred = timelineCred;
+    if (inputInterval === undefined || inputInterval === null) {
```

Comment:
This is the one case where using `==` is preferable to `===`. We can just do:

```js
if (inputInterval == null) {
  this._inputInterval = this._getDefaultInterval(timelineCred);
} else {
  this._inputInterval = inputInterval;
}
```

The following are also fine choices:
```js
this._inputInterval = inputInterval == null 
    ? this._getDefaultInterval(timelineCred) 
    : inputInterval
```
or
```js
this._inputInterval = NullUtil.orElse(
    inputInterval, 
    this._getDefaultInterval(timelineCred)
)
```